tuf/api/metadata.py

```diff
@@ -343,10 +346,26 @@ def __init__(
         expires: datetime,
         unrecognized_fields: Optional[Mapping[str, Any]] = None,
     ) -> None:
+        spec_list = spec_version.split(".")
+        if len(spec_list) != 3 or not all(el.isdigit() for el in spec_list):
+            raise ValueError(
+                f"spec_version must be in semver format, got {spec_version}"
+            )
+        supported_ver = SPECIFICATION_VERSION.split(".")
+        if spec_list[0] != supported_ver[0]:
+            raise exceptions.UnsupportedSpecificationError(
+                f"Unsupported major spec_version, got {spec_list}, "
+                f"supported {supported_ver}"
+            )
```

Comment:
This isn't a major thing and I get that you use different errors because they exist already but will this bring actual value versus code that raises a ValueError -- I don't see anyone ever handling UnsupportedSpecificationError (especially since it'll be caught in serialization code and a DeserializationError will be thrown)?
 
I think this code could just as well be a single if-clause that makes all the checks and a single raise if any of the checks fail -- I think there's just no need to add many separate error messages for something like spec version.

Reply:
Yea, I decided to use `UnsupportedSpecificationError` mostly because it's a specific error.
You have a point that probably nobody takes notice of `UnsupportedSpecificationError`.
I will merge this check into a single if.